src/mentionui.js

```diff
@@ -61,6 +70,55 @@ export default class MentionUI extends Plugin {
 		this._mentionsConfigurations = new Map();
 
 		editor.config.define( 'mention', { feeds: [] } );
+
+		this._requestFeedDebounced = debounce( ( marker, feedText ) => {
+			this._getFeed( marker, feedText )
+				.catch( error => {
+					if ( error.feedDiscarded ) {
+						// Do nothing on discarded feeds as they might come after another one was requested.
+						return;
+					}
+
+					// At this point something bad happened - most likely unreachable host or some feed callback error.
+					// So cleanup marker, remove the UI and...
+					this._hideUIAndRemoveMarker();
+
+					// ...log warning.
+					/**
+					 * The callback used for obtaining mention autocomplete feed thrown and error and the mention UI was hidden or
+					 * not displayed at all.
+					 *
+					 * @error mention-feed-callback-error
+					 */
+					log.warn( 'mention-feed-callback-error: Could not obtain mention autocomplete feed.' );
+				} )
+				.then( feed => {
+					// Do nothing if :
+					// - feed was discarded or empty feed was passed.
+					// - if the marker is not in the document - the selection might have already changed.
+					if ( !feed || !this.editor.model.markers.has( 'mention' ) ) {
```

Comment:
Exhibit A.

Why do we get to `.then()` if something went wrong? This should be handled earlier (in `_getFeed()` most likely).

Reply:
Also, since `marker` is available here by its instance, it may be confusing why you need to check it in the markers collection. I guess that that's because something might've changed in the content and this marker might've been removed? But there should be API methods that you use here to check that and those API methods would then clarify why you need to do so. E.g. it would be `checkIfStillInCompletionMode()` or something similar.

Reply:
I need to re-verify that - I do another round of self-review on this tomorrow.